jsonschema/exceptions.py

```diff
@@ -142,12 +159,18 @@ def _contents(self):
         )
         return dict((attr, getattr(self, attr)) for attr in attrs)
 
-    def _matches_type(self):
+    def _matches_type(self) -> bool:
+        if isinstance(self.schema, _utils.Unset):
```

Comment:
I'd prefer adding `type: ignore`s rather than any runtime things here yeah.

It's a programmer error to get here if `schema` is somehow unset.

Reply:
Done!